### DISCUSSION TITLE
[EXTEND WITH JS] What kind of result does findAuthRecordByToken give?

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4053] [Q&A] [answered]
I'm currently developing a system that allows users to join subjects. To avoid client-side functionality, I opted for extending Pocketbase. However, I encountered an issue with the documentation. Specifically, I couldn't find information on what the following code snippet does:
```javascript
const user = await $app.dao().findAuthRecordByToken(token, $app.settings().recordAuthToken.secret)
```
When I attempt to log the `user` object, it appears to be an object. However, the columns related to the auth-user aren't functioning correctly, and I'm receiving a native function error.

Could someone help with this issue?

Thanks,
- Sven

**Comment by ganigeorgiev** [comment, marked as answer]
I don't understand what do you mean by "I'm receiving a native function error"

Also note that all JSVM helpers execute synchronously within their running goroutine, meaning that you don't need `await`.

Or in other words, the below should return the auth record associated with the auth token (if valid and not expired):
```js
const user = $app.dao().findAuthRecordByToken(token, $app.settings().recordAuthToken.secret)
```

If that doesn't help, please elaborate on what you are trying to do and/or provide a more complete code sample.

**Reply by bijsven** [reply]
Thank you for your response!

What i meant: what kind of data do you get from user... Because it is a object I can't get the info, only [object Object]

**Reply by bijsven** [reply]
Just find out the user.id gives the userid back! Thanks!

**Reply by ganigeorgiev** [reply]
`user` is a `Record` model. For accessing the collection schema fields you'll need to use the designated `get()` and `set()` methods. See https://pocketbase.io/docs/js-records/#getset-record-fields.
